staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy_test.go

```diff
@@ -774,6 +779,106 @@ func TestGetContextForNewRequest(t *testing.T) {
+			Groups: []string{"one", "two"},
+		}
+		path := "/request/path"
+		apiService := &apiregistration.APIService{
+			ObjectMeta: metav1.ObjectMeta{Name: "v1.foo"},
+			Spec: apiregistration.APIServiceSpec{
+				Service:  &apiregistration.ServiceReference{Name: "test-service", Namespace: "test-ns", Port: ptr.To(int32(443))},
+				Group:    "foo",
+				Version:  "v1",
+				CABundle: testCACrt,
+			},
+			Status: apiregistration.APIServiceStatus{
+				Conditions: []apiregistration.APIServiceCondition{
+					{Type: apiregistration.Available, Status: apiregistration.ConditionTrue},
+				},
+			},
+		}
+		targetServer := httptest.NewUnstartedServer(target)
+		serviceCert := svcCrt
+		if cert, err := tls.X509KeyPair(serviceCert, svcKey); err != nil {
+			t.Fatal(err)
+		} else {
+			targetServer.TLS = &tls.Config{Certificates: []tls.Certificate{cert}}
+		}
+		targetServer.StartTLS()
+		defer targetServer.Close()
+
+		serviceResolver := &mockedRouter{destinationHost: targetServer.Listener.Addr().String()}
+		handler := &proxyHandler{
+			localDelegate:              http.NewServeMux(),
+			serviceResolver:            serviceResolver,
+			proxyCurrentCertKeyContent: func() ([]byte, []byte) { return emptyCert(), emptyCert() },
+			tracerProvider:             otelTracer,
+		}
+		server := httptest.NewServer(contextHandler(handler, user))
+		defer server.Close()
+
+		handler.updateAPIService(apiService)
+		curr := handler.handlingInfo.Load().(proxyHandlingInfo)
+		handler.handlingInfo.Store(curr)
+		var propagator propagation.TraceContext
+		req, err := http.NewRequest(http.MethodGet, server.URL+path, nil)
+		if err != nil {
+			t.Errorf("%s: %v", name, err)
+			return
+		}
+
+		_, err = http.DefaultClient.Do(req)
+		if err != nil {
+			t.Errorf("%s: %v", name, err)
+			return
+		}
+
+		id, ok := target.headers["Traceparent"]
+		if !ok {
+			t.Errorf("%s: expected traceparent header", name)
+			return
+		}
+
+		h := http.Header{
+			"Traceparent": id,
+		}
+		ctx := propagator.Extract(context.Background(), propagation.HeaderCarrier(h))
+		spanCtx := trace.SpanContextFromContext(ctx)
+		if !spanCtx.IsValid() {
+			t.Errorf("%s: expected valid span context", name)
+			return
+		}
+
+		expectedSpanCtx := fakeRecorder.Ended()[0].SpanContext()
+		if expectedSpanCtx.TraceID() != spanCtx.TraceID() {
+			t.Errorf("%s: expected trace id to match. expected: %v, but got %v", name, expectedSpanCtx.TraceID(), spanCtx.TraceID())
+			return
+		}
+
+		if expectedSpanCtx.SpanID() != spanCtx.SpanID() {
+			t.Errorf("%s: expected span id to match. expected: %v, but got: %v", name, expectedSpanCtx.SpanID(), spanCtx.SpanID())
+			return
+		}
+
+		expectedSpanNames := []string{"HTTP GET", "aggregator"}
+		spanNames := []string{}
+		for _, span := range fakeRecorder.Ended() {
+			spanNames = append(spanNames, span.Name())
+		}
+		if e, a := expectedSpanNames, spanNames; !reflect.DeepEqual(e, a) {
+			t.Errorf("%s: expected %v, got %v", name, e, a)
+			return
+		}
+	})
```

Comment:
The test lacks structure. 100 lines without comments or even better for tests `t.Log` is hard to maintain and hard to read.

Reply:
log lines added in c460e48